JSONFormat.go

```diff
@@ -39,7 +45,7 @@ func (j jsonEntry) MarshalJSON() ([]byte, error) {
 		return nil, ErrNoMessage
 	}
 
-	_, err := buf.WriteString(fmt.Sprintf("{\"%s\":\"%s\", \"%s\":\"%s\", \"%s\":\"%s\"", j.levelKey, j.Level, j.eventKey, j.Event, j.messageKey, j.Message))
+	_, err := buf.WriteString(fmt.Sprintf("{\"%s\":\"%s\", \"%s\":\"%s\", \"%s\":\"%s\"", j.levelKey, j.Level, j.eventKey, j.Event, j.messageKey, escapeSpecialChars(j.Message)))
```

Comment:
Maybe because of my C# background, but to me it's really weird writing such low level code (manually writing the braces in the Json document, handling escaping, even counting when we need to print the trailing comma), and not using a library for that.

For example can't we replace line 47-87 with something like this?
```
	logData := make(map[string]string)
	logData[j.levelKey] = j.Level
	logData[j.eventKey] = j.Event
	logData[j.messageKey] = j.Message
	for k, v := range j.Meta {
		logData[k] = v
	}

	return json.Marshal(logData)
```

So we save ~30 lines of custom logic, and don't have to worry about the escaping either.